configure.ac

```diff
@@ -466,10 +466,8 @@ AC_ARG_ENABLE([coverage],
 AC_ARG_ENABLE([slow-tests],
   [AS_HELP_STRING([--enable-slow-tests],[build and run slow tests])])
 
-if test x"$with_check" = xyes; then
 AC_ARG_ENABLE([syslog-tests],
   [AS_HELP_STRING([--enable-syslog-tests],[build and run syslog tests])])
```

Comment:
ebcff547bc47a6a3c195f6eb9d839ffed60a7381 did. Looks like portability is a concern for installed artifacts only. According to [SO](http://stackoverflow.com/questions/7619607/how-portable-is-linking-executables-against-loadable-modules) and the included [link](http://stackoverflow.com/questions/2339679/what-are-the-differences-between-so-and-dylib-on-osx), only Mac OS X is affected.